CHANGELOG.md

```diff
@@ -145,6 +145,9 @@ Ref: https://keepachangelog.com/en/1.0.0/
     * `WithChainID` / `WithBlockHeight` / `WithBlockHeader` must be used to set values on the context
 * (x/bank) [#17569](https://github.com/cosmos/cosmos-sdk/pull/17569) `BurnCoins` takes an address instead of a module name
 * (x/distribution) [#17670](https://github.com/cosmos/cosmos-sdk/pull/17670) `AllocateTokens` takes `comet.VoteInfos` instead of `[]abci.VoteInfo`
+* (types) [#17738](https://github.com/cosmos/cosmos-sdk/pull/17738) `WithBlockTime()` & `BlockTime()` were deprecated in favor of `WithHeaderInfo()` & `HeaderInfo()`
```

Comment:
Changelog must be updated, setter removed, getter alias time from header info (behavior change) and deprecated.